src/pipeline/mod.rs

```diff
@@ -0,0 +1,213 @@
+#![cfg(feature = "unstable")]
+
+use futures::lazy;
+use prelude::*;
+use iter::internal::*;
+use std::collections::VecDeque;
+use Scope;
+
+mod test;
+
+pub trait IntoParPipeline
+    where Self: Sized + IntoIterator, Self::Item: Send,
+{
+    /// Converts a sequential iterator into a parallel one. The
+    /// sequential iterator will be driven from a single thread, and a
+    /// Rayon task will be created for each item in the iterator.
+    ///
+    /// In general, it is preferable to use a native parallel iterator
+    /// if they are available. It may also be better to collect the
+    /// items into a vector and then use `into_par_iter()` on that
+    /// vector -- this can help us to create fewer parallel tasks and
+    /// reduce overall overhead.
+    ///
+    /// If the sequential iterator is not `Send`, see `into_par_pipeline_scoped()`.
+    fn into_par_pipeline(self) -> Pipeline<Self> where Self: Send;
```

Comment:
Maybe `into_par_stream`?  I think "stream" is a little clearer than "pipeline" in general.